src/Tinker.jl

```diff
@@ -12,24 +12,40 @@ struct Rectangle <: Shape
     y::Number
     w::Number
     h::Number
+    pts::AbstractArray
```

Comment:
Is this to allow a rotated Rectangle? If so, perhaps add an `angle` field. If not, I think it would be better to insist that `Rectangle` is genuinely a rectangle, simply because anything else could be surprising for future users (names are very influential in how you think about code). In that case it would be better to create a new `Shape` subtype.